app/db/repository/authenticator_repository.rb

```diff
@@ -14,7 +20,10 @@ def find_all(type:, account:)
             "#{account}:webservice:conjur/#{type}/%"
           )
         ).all.map do |webservice|
-          load_authenticator(account: account, id: webservice.id.split(':').last, type: type)
+          service_id = service_id_from_id(webservice.id)
+          next unless webservice.id.split(':').last == "conjur/#{type}/#{service_id}"
```

Comment:
It's not clear what business logic this is performing. Can you add a comment elaborating what the outcome of this filtering is?